### PR TITLE
fix(project-access): export 'clearCdsModuleCache ' to allow clear project's cds cache from outside

### PR DESCRIPTION
**Pull request by 815are**
Issue #1975

We noticed issues:
1. app -> https://github.com/SAP-samples/btp-lcnc-capex
2. Please note - this app does not have any annotation in https://github.com/SAP-samples/btp-lcnc-capex/tree/main/app
3. When we call `getCapModelAndService` - then it gets cached following:
image

4. Then after generating new CAP app in `app` folder, calls for `cds.compile` does not return compiled result from newly added app from `app`

1st solution(not in PR anymore - 4b3d877611ee8066cfd9a10067d3f80f1b3439ba):
1. We have hardcoded `modelPaths` -> https://github.com/SAP/open-ux-tools/blob/main/packages/project-access/src/project/cap.ts#L147-L151
2. Validate those paths after `cds.load` calls
3. If modelPath had cached as `undefined`, then delete such cached entry

2nd solution(in PR):
1. export new method `clearCdsModuleCache` with parameter `projectRoot`, which resolve cds module and calls:
```
cds.resolve.cache = {};
```

### CODE REVIEW

File: packages/project-access/src/project/cap.ts [hunk 1]
```diff
@@ -156,6 +175,8 @@ export async function getCapModelAndServices(
     _logger?.info(`@sap-ux/project-access:getCapModelAndServices - Using 'cds.root': ${cds.root}`);
 
     let services = cds.compile.to.serviceinfo(model, { root: _projectRoot }) ?? [];
+    // Validate cds cache for model paths - there can be case when one of folder does not have annotations yet(for example app folder)
+    validateCdsCache(cds, modelPaths);
```
Comment (voicis) [inline packages/project-access/src/project/cap.ts:179]: I do not think it is a good idea to start looking into specific cache values. It's structure or logic how it is interpreted can change at any time and we'll end up having the same or worse issue. 

Clearing the cache directly is somewhat in the grey area already and we shouldn't go further. than that.

**Review by voicis — DISMISSED** [review]
- Code changes adhere to the development guidelines
- Changeset present
- Good test coverage

**Review by voicis — APPROVED** [review]

File: packages/project-access/src/project/cap.ts [hunk 2]
```diff
@@ -425,6 +425,34 @@ async function loadCdsModuleFromProject(capProjectPath: string, strict: boolean
...
+ * Method to clear CAP CDS module cache for passed project path.
+ *
+ * @param projectRoot root of a CAP project.
+ * @returns True if cache cleared successfully.
+ */
+export async function clearCdsModuleCache(projectRoot: string): Promise<boolean> {
+    try {
+        const cds = await loadCdsModuleFromProject(projectRoot);
+        if (cds) {
+            _clearCdsModuleCache(cds);
```
[the remark this answers is not in the document]
Comment (815are) [reply packages/project-access/src/project/cap.ts]: correct - such case was missed. Fix -> 8d3cc9af7b56873abf7e97232197aa608283758d

File: packages/project-access/src/project/cap.ts [hunk 3]
```diff
@@ -425,6 +425,34 @@ async function loadCdsModuleFromProject(capProjectPath: string, strict: boolean
...
+    }
+    return true;
+}
+
+/**
+ * Method to clear CAP CDS module cache for passed cds module.
+ *
+ * @param cds CAP CDS module
+ */
+function _clearCdsModuleCache(cds: CdsFacade): void {
```
[the remark this answers is not in the document]
Comment (815are) [reply packages/project-access/src/project/cap.ts]: it was both - overlap with names and as it is 'private', then I added `_`.
Applied proposed correction -> 8d3cc9af7b56873abf7e97232197aa608283758d

**Review by Klaus-Keller — APPROVED** [review]
Thanks @815are,
- code is straight forward and easy to understand
- changeset exists and explains the changes
- test coverage is great
- did no manual test, just visual review

Approved from my side.

**Review by broksy — APPROVED** [review]
Understandable change and good test coverage.
Did not test manually.
Approved from my side.

**Pull request merged**